docs/intro/tutorial07.rst

```diff
@@ -64,6 +64,25 @@ This will download the pato.db sqlite file once, and cache it.
 
 PyStow is used to cache the file, and the default location is ``~/.data/oaklib``.
 
+By default, a cached SQLite file will be automatically refreshed (downloaded
+again) if it is older than 7 days. That behavior can be controlled with the
+global ``--caching`` option. For example, to force OAK to always download the
+file regardless of its age:
+
+.. code-block::
+
+    runoak --caching=refresh -i sqlite:obo:pato search t~shape
+
+Other possible values for the ``--caching`` option include:
+
+- ``no-refresh`` to prevent OAK from re-downloading the file even it is older
+  than 7 days;
+- ``Xd`` to refresh a cached file older than _X_ days;
+- ``Xw`` to refresh a cached file older than _X_ weeks.
+
+You may also use the ``cache-clear`` command to force clearing any cached
+SQLite file at anytime.
```

Comment:
So `runoak --caching= clear -i sqlite:obo:pato search t~shape` will clear the entire cache, not just delete the pato cached file?

Reply:
Yes. That’s the difference between `--caching=clear` (or `--caching=reset`, the two are equivalent) and `--caching=refresh`.

```sh
$ runoak --caching=refresh -i sqlite:obo:path search t~shape
```

will forcefully re-download `pato.db` regardless of its age, but will not touch any other cached file.

```sh
$ runoak --caching=clear -i sqlite:obo:path search t~shape
```

will delete _all_ cached database files regardless of their age. Then the cache code will find that `pato.db` is not in the cache (it just deleted that file itself!), and so will download it again.

So after a command in which you used `--caching=clear`, any subsequent command that uses another database will need to download it again, even if that subsequent command does not explicitly request a refresh.

Reply:
Note that the `--caching=clear` _option_ is not the same thing as the `cache-clear` _command_.

The `cache-clear` command will forcefully delete any cached database that is older than 100 days (or the number of days specified by the `--days-old` option), regardless of any caching policy.

So, the equivalent of the `--caching=clear` option would be `clear-cache --days-old=0` (remove any file “older than 0 day“), except that `--caching=clear` option would only take effect _when a command actually tries to use the cache_ (e.g. a command involving a `sqlite:obo:...` selector), whereas `clear-cache --days-old=0` would take effect immediately and unconditionally.

(IMO there would no longer be a real need for `clear-cache` if this PR is applied, as that command was merely a crutch for the lack of proper cache management. But that command existed before, so it should be kept.)

Reply:
Sounds great! In this case, I would copy/paste your comments in the tutorial docs so that others understand these distinctions as well!

Reply:
I don’t think a _tutorial_ should be cluttered with exhaustive documentation about all the precise aspects of a given feature.

I put the bit of doc in the `tutorial07.rst` document because unless I missed something that is one of the the two only places in the entire doc where the fact that SQLite DB files can be automatically downloaded and cached somewhere.

The other place is the [Getting Started](https://incatools.github.io/ontology-access-kit/intro/tutorial01.html#query-the-obo-library) document, which says:

> The base command takes an --input (or just -i) option that specifies the input selector. We will return to the full syntax later, but one pattern that is useful to know is sqlite:obo:<ontology-id>, which uses the SQL Database Adapter to fetch and use the SQLite version of an ontology.
> […]
> The first time you run this there will be a lag as the file is downloaded, but after that it will be cached.

And a “Getting Started” document is, I believe, no more an appropriate place for exhaustive documentation than a “tutorial”.

Exhaustive explanation about how caching works and is controlled would rather belong to _reference documentation_, but the mere concept of caching is currently completely absent of OAK’s reference docs, so I had no place where I could add anything.

For example, the reference documentation about the [selectors](https://incatools.github.io/ontology-access-kit/packages/selectors.html) merely has this to example to show:

> `sqlite:obo:go` - pre-made SQLite from Semantic SQL registry

And the reference documentation for the [SQL Database Adapter](https://incatools.github.io/ontology-access-kit/packages/implementations/sqldb.html) does not say anything at all about automatic download and caching.

Reply:
I’ve added a new section in the CLI reference documentation to fully explain how the `--caching` option works.

Reply:
This is perfect, thanks @gouttegd!